tinyxml2.h

```diff
@@ -305,9 +305,9 @@ class DynArray
         if ( cap > _allocated ) {
             TIXMLASSERT( cap <= INT_MAX / 2 );
             const int newAllocated = cap * 2;
-            T* newMem = new T[newAllocated];
+            T* newMem = new T[static_cast<unsigned int>(newAllocated)];

```

Comment:
Is this better fixed by changing the type of `newAllocated`?

Reply:
That would just push the problem to the line 307 as `cap` is int:
`const int newAllocated = cap * 2;`
Changing `cap` from `int` to `unsigned int` changes the function declaration and will most likely cause other warnings...

[also, this suggest that keeping `int` is better](https://google.github.io/styleguide/cppguide.html#Integer_Types)

> On Unsigned Integers
>Unsigned integers are good for representing bitfields and modular arithmetic. Because of historical accident, the C++ standard also uses unsigned integers to represent the size of containers - many members of the standards body believe this to be a mistake, but it is effectively impossible to fix at this point. The fact that unsigned arithmetic doesn't model the behavior of a simple integer, but is instead defined by the standard to model modular arithmetic (wrapping around on overflow/underflow), means that a significant class of bugs cannot be diagnosed by the compiler. In other cases, the defined behavior impedes optimization.
>That said, mixing signedness of integer types is responsible for an equally large class of problems. The best advice we can provide: try to use iterators and containers rather than pointers and sizes, try not to mix signedness, and try to avoid unsigned types (except for representing bitfields or modular arithmetic). Do not use an unsigned type merely to assert that a variable is non-negative.

That being said, I don't have a strong feeling about this. (I just don't like to fight standards.) If you would prefer changing `int` to `unsigned int` I'd do it. 🙂